### ISSUE TITLE
Update for the 1.0 SDK

### ISSUE THREAD

**Issue by rtsuk** [rtsuk/crankstart#1]
The 1.0 SDK made a lot of changes over the 0.11 SDK. Vote and comment on this issue if you like to write a Playdate game in Rust would use an updated Crankstart crate.

**Comment by mysteriouspants** [comment]
Very excited for playdate, very excited to build things for it, and Rust is my preferred language for recreational projects.

**Comment by bleedingpixels** [comment]
I am interested and would like to know what needs to be updated to get CS to work for 1.0.

**Comment by rtsuk** [comment]
https://github.com/rtsuk/crankstart/tree/sdk_1_0 is the work in progress. I've regenerated the bindings and updates the crate to use them. This was enough to get the Life example working in the simulator.

Hello World should be easy, one just has to modify it to use the new way of specifying text style and drawing.

The Sprite example should also be easy, just a few renames and dropping parameters on drawing calls.

I haven't tried Klondike yet.

When trying to build for device, one sees this

```
error: `rust-src` component not found. Run `rustup component add rust-src`.
note: run with `RUST_BACKTRACE=1` for a backtrace
Error: cargo failed with error ExitStatus(ExitStatus(256))
```

but

```
crankstart $ rustup component add --target thumbv7em-none-eabihf rust-src
info: component 'rust-src' is up to date
```

So something with cargo or cargo xbuild has changed in the last few months which needs to be diagnosed and fixed.

**Comment by rtsuk** [comment]
`cargo install cargo-xbuild` updated xbuild and fixed the rust-src issue.

**Comment by bleedingpixels** [comment]
Can't build examples as it is restricted to macOS atm; was using it under windows before. I will root around and see if I can build for windows as I know you don't like supporting windows.

**Comment by rtsuk** [comment]
When you say "it" is restricted, what are you referring to?

**Comment by bleedingpixels** [comment]
I got hello_world to run on device. Like you said some parameters needed to change. As for the restricted part, crankstart doesn't build for the simulator on windows, that was the error crankstart had when I tried to build for the simulator. It is pretty finicky to build and run pdx for c/c++ on the simulator and was undocumented the last time I tried so I am not really worried about it.

I got life to work as you said, and will try the sprite example, then maybe tackle klondike.
